content/blog/2022-03-31-training-and-saving-models-on-a-dedicated-runner.md

```diff
@@ -0,0 +1,375 @@
+simple `RandomForestClassifier()` based on some generated data. We then use the
+model to make some predictions and plot those predictions in a confusion matrix.
+
+While running the script the model is kept in memory, meaning it is discarded as
+soon as the script finishes. In order to save the model for later, we need to
+dump it as a binary file. We do so with `joblib.dump()`, and later we can read
+the model using `joblib.read()`.
+
+> 💡 You can also use `pickle.dump()` if you prefer.
+
+The outputs of `train.py` are:
+
+- `metrics.txt`: a file containing metrics on model performance;
+- `confusion_matrix.png`: a summary of the prediction results of our model;
+- `random_forest.joblib`: the binary output of the trained model.
+
+All of these files are saved to the `model` directory on the AWS runner.
+
+```python
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import plot_confusion_matrix
+import matplotlib.pyplot as plt
+import json
+import os
+import joblib
+import numpy as np
+
+# Read in data
+X_train = np.genfromtxt("data/train_features.csv")
+y_train = np.genfromtxt("data/train_labels.csv")
+X_test = np.genfromtxt("data/test_features.csv")
+y_test = np.genfromtxt("data/test_labels.csv")
+
+# Fit a model
+depth = 5
+clf = RandomForestClassifier(max_depth=depth)
+clf.fit(X_train,y_train)
+
+# Calculate accuracy
+acc = clf.score(X_test, y_test)
+print(acc)
+
+# Create model folder if it does not yet exist
+if not os.path.exists('model'):
+    os.makedirs('model')
+
+# Write metrics to file
+with open("model/metrics.txt", 'w+') as outfile:
+        outfile.write("Accuracy: " + str(acc) + "\n")
+
+# Plot confusion matrix
+disp = plot_confusion_matrix(clf, X_test, y_test, normalize='true',cmap=plt.cm.Blues)
+plt.savefig('model/confusion_matrix.png')
+
+# Save the model
+joblib.dump(clf, "model/random_forest.joblib")
+```
+
+# Train the model on a runner and export it
+
+Now that we have a script to train our model and save it as a file, let’s set up
+our CI/CD to provision a runner and run the script. We define our workflow in
+`cml.yaml` and save it in the `.github/workflows` directory. This way Github
+will automatically go through the workflow whenever it is triggered. In this
+case the triggers are on push and on a daily schedule.
+
+> 💡 The name of the workflow doesn’t matter, as long as it’s a `.yaml` and
+> located in the `.github/workflows` directory. You can have multiple workflows
+> in there as well.
+> ([Documentation](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions))
+
+The workflow we defined first provisions a runner on AWS and then uses that
+runner to train the model. Once we have the outputs saved, we want to export
+them out of the runner. That way we can simply discard the runner once we are
+done training.
+
+In this guide we will explore two options to export the model from the runner:
```

Comment:
This phrasing is a little confusing because it sounds like we're starting a new guide instead of continuing the existing one. Maybe say something like, "In this section, ..."